packages/mobile/src/home/CashInBottomSheet.tsx

```diff
@@ -26,13 +43,59 @@ function CashInBottomSheet() {
     setModalVisible(false)
   }
 
+  const asyncProviders = useAsync(async () => {
+    if (!account) {
+      Logger.error(TAG, 'No account set')
+      setModalVisible(true)
+      return
+    }
+    // Use cEUR if that is their local currency, otherwise default to cUSD
+    const currencyToBuy =
+      localCurrency === LocalCurrencyCode.EUR ? CiCoCurrency.CEUR : CiCoCurrency.CUSD
+
+    try {
+      const providers = await fetchProviders({
+        userLocation,
+        walletAddress: account,
+        fiatCurrency: localCurrency,
+        digitalAsset: currencyToBuy,
+        fiatAmount: 20,
+        digitalAssetAmount: 20,
+        txType: 'buy',
+      })
+      setModalVisible(true)
```

Comment:
For correctness, we should not be calling these setters within the async code.
The component may be unmounted by the time they are called.
We can pass [`onSuccess`/`onError` options to `react-async-hook`](https://github.com/slorber/react-async-hook/blob/f8569186fbf77607b0f6d46a62639cd838b4d966/test/useAsync.test.ts#L43-L44) to trigger the needed state updates.